crates/ty_python_semantic/resources/mdtest/snapshots/union_call.md_-_Calling_a_union_of_function_types_-_A_smaller_scale_example.snap

```diff
@@ -0,0 +1,70 @@
+---
+source: crates/ty_test/src/lib.rs
+expression: snapshot
+---
+---
+mdtest name: union_call.md - Calling a union of function types - A smaller scale example
+mdtest path: crates/ty_python_semantic/resources/mdtest/diagnostics/union_call.md
+---
+
+# Python source files
+
+## mdtest_snippet.py
+
+```
+ 1 | def f1() -> int:
+ 2 |     return 0
+ 3 | 
+ 4 | def f2(name: str) -> int:
+ 5 |     return 0
+ 6 | 
+ 7 | def _(flag: bool):
+ 8 |     if flag:
+ 9 |         f = f1
+10 |     else:
+11 |         f = f2
+12 |     # error: [too-many-positional-arguments]
+13 |     # error: [invalid-argument-type]
+14 |     x = f(3)
+```
+
+# Diagnostics
+
+```
+error: lint:invalid-argument-type: Argument to this function is incorrect
```

Comment:
I think one thing that would make the concise diagnostic for `invalid-argument-type` more informative (and less duplicate-looking in some cases) would be if we name the target function here, like we do below for `too-many-positional-arguments` (and for many of the other call diagnostics)

Reply:
Nice. This is done. And _super_ annoying to update mdtest for. Hah. But I think the messages are better.